### PR TITLE
feat(sdk-core): refactor signConvert to steps (3/4)

### PR DESCRIPTION
**Pull request by zahin-mohammad**
## Description

This PR splits up `signConvert` into `signConvertStep1`, `signConvertStep2`, and `signConvertStep3`. 

Each step is fully typed out with typed inputs and outputs, and the implementation removes the use of any types and castings. 

The previous `signConvert` method is marked as deprecated. 

This PR is needed before introducing paillier proofs to the signing flow. The paillier proofs require updating the signConvert method which in its previous form was too complicated and error prone. 

## Issue Number

BG-78338

## Type of change

Please delete options that are not relevant.

- [x] Refactor

# How Has This Been Tested?

Existing code has been updated to use the new steps. All tests are passing.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code compiles correctly for both Node and Browser environments
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [x] The ticket or github issue was included in the commit message as a reference
- [x] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts [hunk 1]
```diff
@@ -380,12 +388,555 @@ export default class Ecdsa {
...
+    };
+  }
+
+  /**
+   * Perform multiplicitive-to-additive (MtA) share conversion with another signer.
+   * Connection 1.2 in https://lucid.app/lucidchart/7061785b-bc5c-4002-b546-3f4a3612fc62/edit?page=IAVmvYO4FvKc#
+   * @param {SignConvert} shares
+   * @returns {SignConvertRT}
+   */
+  async signConvertStep1(shares: SignConvertStep1): Promise<SignConvertStep1Response> {
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts:403]: Note: these methods can be further optimized in future PR's. 

This PR just splits up the behemoth that is signConvert, without optimizing those functions further.

File: modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts [hunk 2]
```diff
@@ -121,17 +130,16 @@ export async function createUserSignShare(xShare: XShareWithNtilde, yShare: YSha
  * Creates the Gamma Share and MuShare with User WShare and AShare From BitGo
  * @param {WShare} wShare User WShare
  * @param {AShare} aShare AShare from Bitgo
- * @returns {Promise<CreateUserGammaAndMuShareRT>}
+ * @returns {Promise<SignConvertStep2Response>}
  */
-export async function createUserGammaAndMuShare(wShare: WShare, aShare: AShare): Promise<CreateUserGammaAndMuShareRT> {
+export async function createUserGammaAndMuShare(wShare: WShare, aShare: AShare): Promise<SignConvertStep2Response> {
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts:135]: TODO: we can remove this fn in the future and add this validation to `signConvertStep2`

File: modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts [hunk 3]
```diff
@@ -599,7 +599,7 @@ describe('TSS Ecdsa Utils:', async function () {
       const [userXShare, bitgoXShare] = await Promise.all([
         MPC.appendChallenge(userSigningKey.xShare, serializedEntChallenge),
```
Comment (rlutzbitgo) [inline modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts:600]: If `appendChallenge()` is no longer asynchronous, do we need to `await Promise.all()`?

Comment (zahin-mohammad) [reply modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts:600]: good catch, its no longer needed

File: modules/bitgo/test/v2/unit/tss/ecdsa.ts [hunk 4]
```diff
@@ -307,9 +307,9 @@ describe('Ecdsa tss helper functions tests', function () {
       });
 
       it('should fail if the Xshare doesnt belong to the User', async function () {
-        let xShare = await mpc.appendChallenge(userKey.xShare, mockSerializedChallengeWithProofs);
+        let xShare = mpc.appendChallenge(userKey.xShare, mockSerializedChallengeWithProofs);
         xShare = { ...xShare, i: 3 };
-        const yShare: ECDSAMethodTypes.YShareWithNtilde = await mpc.appendChallenge(userKey.yShares[3], {
+        const yShare: ECDSAMethodTypes.YShareWithNtilde = mpc.appendChallenge(userKey.yShares[3], {
```
Comment (rlutzbitgo) [inline modules/bitgo/test/v2/unit/tss/ecdsa.ts:312]: Same with here

File: modules/bitgo/test/v2/unit/tss/ecdsa.ts [hunk 5]
```diff
@@ -280,8 +280,8 @@ describe('Ecdsa tss helper functions tests', function () {
 
     describe('createUserSignShare:', async function () {
       it('should succeed to create User SignShare', async function () {
-        const xShare = await mpc.appendChallenge(userKey.xShare, mockSerializedChallengeWithProofs);
-        const yShare: ECDSAMethodTypes.YShareWithNtilde = await mpc.appendChallenge(userKey.yShares[3], { ntilde: xShare.ntilde, h1: xShare.h1, h2: xShare.h2 });
+        const xShare = mpc.appendChallenge(userKey.xShare, mockSerializedChallengeWithProofs);
+        const yShare: ECDSAMethodTypes.YShareWithNtilde = mpc.appendChallenge(userKey.yShares[3], { ntilde: xShare.ntilde, h1: xShare.h1, h2: xShare.h2 });
```
[the remark this answers is not in the document]
Comment (zahin-mohammad) [reply modules/bitgo/test/v2/unit/tss/ecdsa.ts]: yup, fixed..

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/types.ts [hunk 6]
```diff
@@ -197,58 +191,90 @@ export interface GShare {
...
+};
+
+export type SignConvertStep1 = {
+  xShare: XShareWithChallenges; // XShare of the current participant
+  yShare: YShare; // YShare corresponding to the other participant
+  kShare: KShare; // share to be modified and sent to other participant
+};
+
+export type SignConvertStep1Response = {
+  bShare: BShare; // participant share
```
Comment (rlutzbitgo) [inline modules/sdk-core/src/account-lib/mpc/tss/ecdsa/types.ts:213]: Kind of an ultra-nit, but for people not familiar with the ECDSA TSS signing flow, they might not immediately understand that "participant share" == "private share not to be given to anyone else"

It is called out in the type just above, so I think it's fine

**Review by rlutzbitgo — APPROVED** [review]
Looks good to me!

**Review by margueriteblair — APPROVED** [review]
lgtm

**Review by margueriteblair — APPROVED** [review]
lgtm

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts [hunk 7]
```diff
@@ -380,12 +386,529 @@ export default class Ecdsa {
...
+    const xShare = shares.xShare; // currentParticipant secret xShare
+    const yShare: YShareWithChallenges = {
+      ...shares.yShare,
+      ntilde: shares.kShare.ntilde,
+      h1: shares.kShare.h1,
+      h2: shares.kShare.h2,
+    };
+    const signShare = await this.signShare(xShare, yShare);
+    const receivedKShare = shares.kShare;
```
[the remark this answers is not in the document]
Comment (zahin-mohammad) [reply modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts:410]: Addressing in https://github.com/BitGo/BitGoJS/pull/3602

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/types.ts (hunk 6 above)
Comment (zahin-mohammad) [reply modules/sdk-core/src/account-lib/mpc/tss/ecdsa/types.ts:213]: Addressing in https://github.com/BitGo/BitGoJS/pull/3602

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts [hunk 8]
```diff
@@ -380,12 +386,529 @@ export default class Ecdsa {
...
+      wShare,
+      kShare,
+    };
+  }
+
+  /**
+   * Perform multiplicitive-to-additive (MtA) share conversion with another signer.
+   * Connection 1.2 in https://lucid.app/lucidchart/7061785b-bc5c-4002-b546-3f4a3612fc62/edit?page=IAVmvYO4FvKc#
+   * @param {SignConvert} shares
+   * @returns {SignConvertRT}
```
Comment (islamaminBitGo) [inline modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts:399]: could benefit from a comment that signConvertStep1 and signConvertStep2 are not completed by the same party. maybe an overall comment on if we have SignerA starting first and SignerB going second, which functions would each of them call.

**Review by islamaminBitGo — APPROVED** [review]
Great work on splitting signConvert into modularized steps , this will help us a lot in the future!

**Pull request merged**